sherpa/astro/utils/tests/test_astro_utils_unit.py

```diff
@@ -220,3 +224,362 @@ def test_do_group_single_group(func, expected):
 
     ans = do_group([6, 4, 2, 1, -1, 3], [1, -1, -1, -1, -1, -1], func)
     assert ans == pytest.approx([expected])
+
+
+def make_data(data_class):
+    """Create a test data object of the given class.
+
+    Using a string means it is easier to support the various PHA
+    "types" - eg basic, grouping, grouping+quality.
+
+    """
+
+    x0 = np.asarray([1, 3, 7, 12])
+    y = np.asarray([2, 3, 4, 5])
+    if data_class == "1d":
+        return Data1D('x1', x0, y)
+
+    if data_class == "1dint":
+        return Data1DInt('xint1', x0, np.asarray([3, 5, 8, 15]), y)
+
+    chans = np.arange(1, 5)
+    if data_class == "pha":
+        return DataPHA('pha', chans, y)
+
+    # We want to provide PHA tests that check out the grouping and
+    # quality handling (but it is not worth trying all different
+    # variants), so we have "grp" for grouping and no quality [*], and
+    # "qual" for groupbing and quality.
```

Comment:
```suggestion
    # "qual" for grouping and quality.
```

Reply:
I had missed this when @hamogu pointed it out in https://github.com/sherpa/sherpa/pull/1440#discussion_r806213780 - grrr

Reply:
I'm glad that the only thing that's still left to worry about is a comment in a test.